templates/server-statefulset.yaml

```diff
@@ -41,17 +41,21 @@ spec:
       terminationGracePeriodSeconds: 10
       serviceAccountName: {{ template "vault.fullname" . }}
       securityContext:
+        readOnlyRootFilesystem: true
+        runAsNonRoot: true
+        runAsGroup: 1000
+        runAsUser: 100
```

Comment:
One thing I just realized here is if someone overrides our docker image with their own the UID and GID may differ, so we may want to either document that in the docs or comment inline here, or consider making this a template variable.

Reply:
I can make this configurable, good point @malnick !